src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx

```diff
@@ -171,13 +172,16 @@ export class ProsemirrorAdapter {
                         'max-height': 'inherit',
                     }}
                 >
-                    <limel-text-editor-link-menu
-                        link={this.link}
-                        isOpen={this.isLinkMenuOpen}
-                        onLinkChange={this.handleLinkChange}
-                        onCancel={this.handleCancelLinkMenu}
-                        onSave={this.handleSaveLinkMenu}
-                    />
+                    {this.isLinkMenuOpen && (
+                        <limel-text-editor-link-menu
+                            link={this.link}
+                            isOpen={this.isLinkMenuOpen}
+                            onLinkChange={this.handleLinkChange}
+                            onCancel={this.handleCancelLinkMenu}
+                            onSave={this.handleSaveLinkMenu}
+                        />
+                    )}
+                    ,
                 </limel-menu-surface>
             </limel-portal>,
```

Comment:
for the records, I tired getting rid of both `portal` and `menu-surface`, in the hope that `delegateFocus` would work, and that we may not need too complex scripts. This was the code:
```tsx
    public render() {
        return [
            <div id="editor" />,
            <limel-action-bar
                accessibleLabel="Toolbar"
                actions={this.actionBarItems}
                onItemSelected={this.handleActionBarItem}
            />,
            <limel-text-editor-link-menu
                style={{
                    display: this.isLinkMenuOpen ? 'flex' : 'none',
                }}
                link={this.link}
                isOpen={this.isLinkMenuOpen}
                onLinkChange={this.handleLinkChange}
                onCancel={this.handleCancelLinkMenu}
                onSave={this.handleSaveLinkMenu}
            />,
        ];
    }
```
and 
```scss
:host(limel-text-editor-link-menu) {
    position: absolute;
    z-index: 2;
    inset: 1.5rem 0 0 0;
    margin: auto;

    // display: flex; // this is set as an inline style in the component
    flex-direction: column;
    gap: 0.5rem;
    padding: 0.5rem;

    width: calc(100% - 2rem);
    height: fit-content;
    overflow: hidden auto;

    border-radius: 0.5rem;
    background-color: var(--lime-elevated-surface-background-color);
    box-shadow: var(--shadow-depth-16);
}

.actions {
    display: flex;
    justify-content: end;
    gap: 0.5rem;
}
```
<img width="439" alt="image" src="https://github.com/Lundalogik/lime-elements/assets/35954987/83a4c245-958d-4225-a05b-d332a8d44832">

However, this didn't really help. Because:
1. the link popover was always center aligned, but it would stretch out of the container, resulting in bad placement in the rest of the UI (e.g. when the editor is not so tall and is place at the bottom of the screen).
2. didn't solve the `delegateFocus` issue

Reply:
Now I'm gonna try to see if we can make it without `limel-menu-surface` but with the `portal`

Reply:
Thanks for looking into that @Kiarokh